android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -1765,9 +1769,16 @@ public void callInjectedJavaScriptBeforeContentLoaded() {
     public void onMessage(String message) {
       ReactContext reactContext = (ReactContext) this.getContext();
       InternalWebView mContext = this;
-
-      if (mRNCWebViewClient != null) {
-        WebView webView = this;
+      WebView webView = this;
+
+      if (webViewKey != null && mRNCWebViewClient != null) {
+        reactContext.runOnUiQueueThread(() -> {
+          WritableMap data = mRNCWebViewClient.createWebViewEvent(webView, webView.getUrl());
+          data.putString("webViewKey", webViewKey);
+          data.putString("data", message);
+          reactContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class).emit("onMessage", data);
```

Comment:
So I learned that the event names need to be unique for your entire app, I think we should rename this to `ReactNativeWebViewOnMessage` or something of that sort.

https://github.com/facebook/react-native/blob/main/Libraries/EventEmitter/NativeEventEmitter.js#L35

I'll rename both in a follow up PR for both iOS and Android

Reply:
https://app.asana.com/0/1201254368985004/1202916067056138/f